### ISSUE TITLE
Option to allow strings for "var-name-mixedcase"

### ISSUE THREAD

**Issue by PaulRBerg** [protofire/solhint#663]
We frequently want to define strings like so, and they act as immutables:

```solidity
string internal RESULTS_DIR;

contructor() {
  RESULTS_DIR = "foo";
}
```

Solhint throws this linting warning:

```
  23:5   warning  Variable name must be in mixedCase                             var-name-mixedcase
```

So end up having to disable `var-name-mixedcase`, although we still want to use it for the other variable types.

It would be helpful to be able to exempt strings from this rule.

**Comment by dbale-altoros** [comment]
hello @PaulRBerg thanks for commenting this

I'm not sure I understand. I mean, you cannot define that variable as a constant ?
But you want to still use naming notation as a constant ?

**Comment by dbale-altoros** [comment]
@PaulRBerg 
the thing is, there is no clear style guide for immutable, at least not in [soliditylang](https://docs.soliditylang.org/en/latest/style-guide.html)

I think there is an overlapping issue between two rules
`immutable-vars-naming` and `var-name-mixedcase`
I need to disable `var-name-mixedcase` for immutable variables
And you can use then `immutable-vars-naming` to config how you want the immutable variables to be treated

```
'immutable-vars-naming': [ 'warn', { immutablesAsConstants: true } ],
"var-name-mixedcase": "error"
```

I'll do it but since yesterday we released a new version, this will not be available until next one... I suppose a couple of weeks.

Thanks for always contributing!

**Comment by dbale-altoros** [comment]
@PaulRBerg 
sorry this is weird
I just checked and that is already done... (i already did that)

![Image](https://github.com/user-attachments/assets/8e04706a-6b0d-4697-b998-ae38113ce486)

So, are you sure you're getting an error defining the variable as immutable ?

**Comment by dbale-altoros** [comment]
I understand the issue now
sorry the two previous messages

**Comment by dbale-altoros** [comment]
I can think of putting a config flag in this rule to skip string variables 
But I'm not sure it is a good thing because it will apply for all string variables of the contract

I need to think about this.. seems like a very particular issue...

**Comment by PaulRBerg** [comment]
Actually, I realize now that it isn't just strings. Any sort of dynamic type that cannot be declared as `constant` or `immutable` *but* which is defined exclusively in the constructor falls in the same category.

What if we allow users to pass specific name globs in the Solhint config that are excluded from the rule? e.g. all variables prefixed as `RESULTS_`

If that isn't possible, I agree with you that there isn't any simple solution for this issue, and we can close the issue.

**Comment by dbale-altoros** [comment]
I'll take a look and see what can i do @PaulRBerg

**Comment by PaulRBerg** [comment]
@dbale-altoros I was referring to user-provided prefixes, so in our case, we would apply `RESULTS_`

But yeah if there is a general catch-all prefix hardcode din Solhint, `IMM_` should work!

**Comment by dbale-altoros** [comment]
ok, I added this fix in this PR
https://github.com/protofire/solhint/pull/680

prefix will be configurable for all non Constants and non Immutable variables to skip the mixedcase variable check, leaving `IMM_` as default but any string value can be put 

if variable has this prefix, this rule will not applied to avoid 
`23:5   warning  Variable name must be in mixedCase                             var-name-mixedcase`
And leaving the room to put things like this
`string private IMM_MESSGAE = "Hope this works!"`

**Comment by PaulRBerg** [comment]
awesome, tyvm @dbale-altoros

@smol-ninja @andreivladbrg let's implement this in the Benchmarks repo
